data/types/food/ingredient_test.go

```diff
@@ -0,0 +1,357 @@
+package food_test
+
+import (
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/ginkgo/extensions/table"
+	. "github.com/onsi/gomega"
+
+	dataNormalizer "github.com/tidepool-org/platform/data/normalizer"
+	"github.com/tidepool-org/platform/data/types/food"
+	testDataTypes "github.com/tidepool-org/platform/data/types/test"
+	testErrors "github.com/tidepool-org/platform/errors/test"
+	"github.com/tidepool-org/platform/pointer"
+	"github.com/tidepool-org/platform/structure"
+	structureValidator "github.com/tidepool-org/platform/structure/validator"
+	"github.com/tidepool-org/platform/test"
+)
+
+func NewIngredient(ingredientArrayDepth int) *food.Ingredient {
+	datum := food.NewIngredient()
+	datum.Amount = NewAmount()
+	datum.Brand = pointer.String(test.NewText(1, 100))
+	datum.Code = pointer.String(test.NewText(1, 100))
+	datum.Ingredients = NewIngredientArray(ingredientArrayDepth)
+	datum.Name = pointer.String(test.NewText(1, 100))
+	datum.Nutrition = NewNutrition()
+	return datum
+}
+
+func CloneIngredient(datum *food.Ingredient) *food.Ingredient {
+	if datum == nil {
+		return nil
+	}
+	clone := food.NewIngredient()
+	clone.Amount = CloneAmount(datum.Amount)
+	clone.Brand = test.CloneString(datum.Brand)
+	clone.Code = test.CloneString(datum.Code)
+	clone.Ingredients = CloneIngredientArray(datum.Ingredients)
+	clone.Name = test.CloneString(datum.Name)
+	clone.Nutrition = CloneNutrition(datum.Nutrition)
+	return clone
+}
+
+func NewIngredientArray(ingredientArrayDepth int) *food.IngredientArray {
+	var datum *food.IngredientArray
+	if ingredientArrayDepth--; ingredientArrayDepth > 0 {
+		datum = food.NewIngredientArray()
+		for count := 0; count < test.RandomIntFromRange(1, 3); count++ {
+			*datum = append(*datum, NewIngredient(ingredientArrayDepth))
+		}
+	}
+	return datum
+}
+
+func CloneIngredientArray(datumArray *food.IngredientArray) *food.IngredientArray {
+	if datumArray == nil {
+		return nil
+	}
+	clone := food.NewIngredientArray()
+	for _, datum := range *datumArray {
+		*clone = append(*clone, CloneIngredient(datum))
+	}
+	return clone
+}
+
+var _ = Describe("Ingredient", func() {
+	It("IngredientBrandLengthMaximum is expected", func() {
+		Expect(food.IngredientBrandLengthMaximum).To(Equal(100))
+	})
+
+	It("IngredientCodeLengthMaximum is expected", func() {
+		Expect(food.IngredientCodeLengthMaximum).To(Equal(100))
+	})
+
+	It("IngredientNameLengthMaximum is expected", func() {
+		Expect(food.IngredientNameLengthMaximum).To(Equal(100))
+	})
+
+	Context("ParseIngredient", func() {
+		// TODO
```

Comment:
why later??

Reply:
All of the parsing tests will be updated when I revamp the parsing to use the `structure` parser (as opposed to the `data`-specific parser. Will add tests with that work.